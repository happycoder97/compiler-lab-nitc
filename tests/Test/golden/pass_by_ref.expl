type Foo {
    int a;
    Foo foo;
}

int setFoo(Foo a) {
    a = alloc();
    a.a = 1234;
    a.foo = alloc();
    a.foo.a = 9988;
    return 0;
}

int main() {
   initialize();
   Foo foo;
   setFoo(foo);
   write(foo.a);
   write(foo.foo.a);
   return 0;
}
