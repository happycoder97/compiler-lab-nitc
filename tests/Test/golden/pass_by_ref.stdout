1234
9988
